caravel/views.py

```diff
@@ -1964,6 +1964,38 @@ def cached_key(self, key):
         return "nope"
 
     @has_access_api
+    @expose("/results/<key>/")
+    @log_this
+    def results(self, key):
+        """Serves a key off of the results backend"""
+        blob = results_backend.get(key)
+        if blob:
+            json_payload = zlib.decompress(blob)
+            obj = json.loads(json_payload)
+            db_id = obj['query']['dbId']
+            session = db.session()
+            mydb = session.query(models.Database).filter_by(id=db_id).first()
```

Comment:
s/first()/one()
it will fail if the DB was removed and will provide meaningful error message